include/pisa/cursor/scored_cursor.hpp

```diff
@@ -7,13 +7,41 @@
 
 namespace pisa {
 
-template <typename Index>
-struct scored_cursor {
-    using enum_type = typename Index::document_enumerator;
-    scored_cursor() = delete;
-    enum_type docs_enum;
-    float q_weight;
-    term_scorer_t scorer;
+template <typename Cursor>
+class ScoredCursor {
+  public:
+    using base_cursor_type = Cursor;
+
+    ScoredCursor(
+        Cursor cursor,
+        TermScorer term_scorer,
+        float query_weight,
+        float max_score = std::numeric_limits<float>::max())
```

Comment:
Why a scored cursor should have a maxscore?

I would just extend a scored cursor and add a maxscore to a new class

Reply:
The idea is to cut down on compilation time, but now that I think of it, it won't help because we generally use one type of cursor for each algorithm, so we'd just replace one template generation with another. I'll change it then.